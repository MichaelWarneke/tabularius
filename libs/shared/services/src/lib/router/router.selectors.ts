import { createFeatureSelector, createSelector } from '@ngrx/store';
import { RouterReducerState } from '@ngrx/router-store';
import { RouterStateUrl } from './router.state';

export const selectRouterState = createFeatureSelector<
  RouterReducerState<RouterStateUrl>
>('router');

export const selectIdOfRoute = createSelector(
  selectRouterState,
  state =>
    state &&
    state.state &&
    state.state.params &&
    (state.state.params.id as string | null)
);
