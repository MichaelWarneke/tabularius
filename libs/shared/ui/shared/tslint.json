{
  "extends": "../../../../tslint.json",
  "rules": {
    "directive-selector": [true, "attribute", "tabuSharedUi", "camelCase"],
    "component-selector": [true, "element", "tabu-shared-ui", "kebab-case"]
  }
}
