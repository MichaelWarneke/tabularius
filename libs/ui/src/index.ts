export { TabuAppSidenavModule } from './lib/nav';
