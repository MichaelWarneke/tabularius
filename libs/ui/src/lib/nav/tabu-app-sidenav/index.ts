export { TabuAppSidenavModule } from './tabu-app-sidenav.module';
