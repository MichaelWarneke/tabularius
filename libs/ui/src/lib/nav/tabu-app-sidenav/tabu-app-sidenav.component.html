<mat-sidenav-container class="sidenav-container">
  <mat-sidenav
    #drawer
    class="sidenav"
    fixedInViewport="true"
    [attr.role]="(isHandset$ | async) ? 'dialog' : 'navigation'"
    [mode]="(isHandset$ | async) ? 'over' : 'side'"
    [opened]="!(isHandset$ | async)"
  >
    <mat-toolbar>Menu</mat-toolbar>
    <mat-nav-list>
      <a
        routerLink="/dash"
        routerLinkActive="active"
        mat-list-item
        href="#"
        i18n
        (click)="closeDrawer()"
        >Dash</a
      >
      <a
        routerLink="/journal"
        routerLinkActive="active"
        mat-list-item
        href="#"
        i18n
        (click)="closeDrawer()"
        >Journal</a
      >
      <a
        routerLink="/reports"
        routerLinkActive="active"
        mat-list-item
        href="#"
        i18n
        (click)="closeDrawer()"
        >Reports</a
      >
      <a
        routerLink="/import"
        routerLinkActive="active"
        mat-list-item
        href="#"
        i18n
        (click)="closeDrawer()"
        >Import</a
      >
      <a
        routerLink="/setup"
        routerLinkActive="active"
        mat-list-item
        href="#"
        i18n
        (click)="closeDrawer()"
        >Setup</a
      >
    </mat-nav-list>
  </mat-sidenav>
  <mat-sidenav-content>
    <mat-toolbar color="primary">
      <button
        type="button"
        aria-label="Toggle sidenav"
        mat-icon-button
        (click)="drawer.toggle()"
        *ngIf="(isHandset$ | async)"
      >
        <mat-icon aria-label="Side nav toggle icon">menu</mat-icon>
      </button>
      <span i18n>TABULARIUS</span> <span>{{ userName }}</span>
      <button
        type="button"
        routerLink="/account"
        aria-label="Toggle sidenav"
        mat-icon-button
      >
        <mat-icon aria-label="Side nav toggle icon">account_circle</mat-icon>
      </button>
    </mat-toolbar>
    <!-- Add Content Here -->
    <router-outlet></router-outlet>
  </mat-sidenav-content>
</mat-sidenav-container>
