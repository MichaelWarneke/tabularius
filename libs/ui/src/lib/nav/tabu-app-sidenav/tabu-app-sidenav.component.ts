import {
  Component,
  OnInit,
  ViewChild,
  Input,
  ChangeDetectionStrategy
} from '@angular/core';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { Breakpoints, BreakpointObserver } from '@angular/cdk/layout';
import { MatSidenav } from '@angular/material';

@Component({
  selector: 'tabu-app-sidenav',
  templateUrl: './tabu-app-sidenav.component.html',
  styleUrls: ['./tabu-app-sidenav.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class TabuAppSidenavComponent implements OnInit {
  isHandset$: Observable<boolean> = this.breakpointObserver
    .observe(Breakpoints.Handset)
    .pipe(map(result => result.matches));

  @ViewChild('drawer') sidenav?: MatSidenav;
  @Input() userName: string | null | undefined;
  @Input() isAuth = false;

  constructor(private breakpointObserver: BreakpointObserver) {}

  ngOnInit() {}

  closeDrawer() {
    if (this.breakpointObserver.isMatched(Breakpoints.Handset)) {
      if (this.sidenav) {
        this.sidenav.close();
      }
    }
  }
}
