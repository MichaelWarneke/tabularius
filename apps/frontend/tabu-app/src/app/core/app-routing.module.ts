import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

const routes: Routes = [
  {
    path: 'dash',
    loadChildren: '@tabularius/modules/app-dash#AppDashModule'
  },
  {
    path: 'journal',
    loadChildren: '@tabularius/modules/app-journal#AppJournalModule'
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: []
})
export class AppRoutingModule {}
