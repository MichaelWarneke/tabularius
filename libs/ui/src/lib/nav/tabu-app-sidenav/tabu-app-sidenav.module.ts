import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TabuAppSidenavComponent } from './tabu-app-sidenav.component';
import { MaterialModule } from '@tabularius/shared-libs';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [TabuAppSidenavComponent],
  imports: [CommonModule, RouterModule, MaterialModule],
  exports: [TabuAppSidenavComponent]
})
export class TabuAppSidenavModule {}
