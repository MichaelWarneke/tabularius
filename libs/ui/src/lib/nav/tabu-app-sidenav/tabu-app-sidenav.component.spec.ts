import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

import { TabuAppSidenavComponent } from './tabu-app-sidenav.component';
import { RouterTestingModule } from '@angular/router/testing';
import { SharedModule } from '@tabularius/shared/ui';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('NavComponent', () => {
  let component: TabuAppSidenavComponent;
  let fixture: ComponentFixture<TabuAppSidenavComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [TabuAppSidenavComponent],
      imports: [NoopAnimationsModule, RouterTestingModule, SharedModule],
      providers: [],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TabuAppSidenavComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should compile', () => {
    expect(component).toBeTruthy();
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
