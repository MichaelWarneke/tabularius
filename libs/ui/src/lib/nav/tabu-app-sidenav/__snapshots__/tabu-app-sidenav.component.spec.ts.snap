// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavComponent should render 1`] = `
<tabu-app-sidenav
  breakpointObserver={[Function BreakpointObserver]}
  isAuth="false"
  isHandset$={[Function Observable]}
  sidenav={[Function MatSidenav]}
>
  <mat-sidenav-container
    class="sidenav-container mat-drawer-container mat-sidenav-container"
  >
    
    <div
      class="mat-drawer-backdrop ng-star-inserted"
    />
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav
      class="sidenav mat-drawer mat-sidenav ng-tns-c7-0 ng-trigger ng-trigger-transform mat-drawer-side mat-sidenav-fixed ng-star-inserted"
      fixedinviewport="true"
      ng-reflect-fixed-in-viewport="true"
      ng-reflect-mode="side"
      ng-reflect-opened="true"
      role="navigation"
      style="top:0px;bottom:0px;transform:none;visibility:visible;0:top;1:bottom;2:transform;3:visibility;"
      tabindex="-1"
    >
      <div
        class="mat-drawer-inner-container"
      >
        <mat-toolbar
          class="mat-toolbar mat-toolbar-single-row"
        >
          Menu
        </mat-toolbar>
        <mat-nav-list
          class="mat-nav-list mat-list-base"
          role="navigation"
        >
          <a
            class="mat-list-item"
            href="/dash"
            mat-list-item=""
            ng-reflect-router-link="/dash"
            ng-reflect-router-link-active="active"
            routerlink="/dash"
            routerlinkactive="active"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/dash"
              />
              <div
                class="mat-list-text"
              />
              Dash
            </div>
          </a>
          <a
            class="mat-list-item"
            href="/journal"
            mat-list-item=""
            ng-reflect-router-link="/journal"
            ng-reflect-router-link-active="active"
            routerlink="/journal"
            routerlinkactive="active"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/journal"
              />
              <div
                class="mat-list-text"
              />
              Journal
            </div>
          </a>
          <a
            class="mat-list-item"
            href="/reports"
            mat-list-item=""
            ng-reflect-router-link="/reports"
            ng-reflect-router-link-active="active"
            routerlink="/reports"
            routerlinkactive="active"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/reports"
              />
              <div
                class="mat-list-text"
              />
              Reports
            </div>
          </a>
          <a
            class="mat-list-item"
            href="/import"
            mat-list-item=""
            ng-reflect-router-link="/import"
            ng-reflect-router-link-active="active"
            routerlink="/import"
            routerlinkactive="active"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/import"
              />
              <div
                class="mat-list-text"
              />
              Import
            </div>
          </a>
          <a
            class="mat-list-item"
            href="/setup"
            mat-list-item=""
            ng-reflect-router-link="/setup"
            ng-reflect-router-link-active="active"
            routerlink="/setup"
            routerlinkactive="active"
          >
            <div
              class="mat-list-item-content"
            >
              <div
                class="mat-list-item-ripple mat-ripple"
                mat-ripple=""
                ng-reflect-disabled="false"
                ng-reflect-trigger="http://localhost/setup"
              />
              <div
                class="mat-list-text"
              />
              Setup
            </div>
          </a>
        </mat-nav-list>
      </div>
    </mat-sidenav>
    <div
      aria-hidden="true"
      class="cdk-visually-hidden cdk-focus-trap-anchor"
    />
    <mat-sidenav-content
      class="mat-drawer-content mat-sidenav-content"
    >
      <mat-toolbar
        class="mat-toolbar mat-primary mat-toolbar-single-row"
        color="primary"
        ng-reflect-color="primary"
      >
        
        <span>
          TABULARIUS
        </span>
        <span>
           false
        </span>
        <tabu-account-menu />
      </mat-toolbar>
      <router-outlet />
    </mat-sidenav-content>
    
  </mat-sidenav-container>
</tabu-app-sidenav>
`;
